tss-esapi/src/context/handle_manager.rs

```diff
@@ -41,8 +41,12 @@ impl HandleManager {
         }
 
         if self.open_handles.contains_key(&handle) {
-            error!("Handle({}) is already open", ESYS_TR::from(handle));
-            return Err(Error::local_error(WrapperErrorKind::InvalidHandleState));
+            // It is safe to call unwrap because the existance of the key has already been verified.
+            let stored_handle_drop_action = self.open_handles.get(&handle).unwrap();
```

Comment:
Why not just fold this with the if as something like:

```rust
if let Some(stored_handle_drop_action) = self.open_handles.get(&handle) {
```

and avoid the `unwrap` altogether? :)

Reply:
Fixed :)